R/postprocess.R

```diff
@@ -1,3 +1,132 @@
+#' Extract samples of the delay distribution parameters
+#'
+#' @param fit A model fit with `epidist::epidist`
+#' @param ... Additional arguments passed to `brms::prepare_predictions`
+#' @family postprocess
+#' @autoglobal
+#' @export
+delay_samples <- function(fit, ...) {
+  args <- list(...)
+  if ("newdata" %in% names(args)) {
+    newdata <- brms:::validate_newdata(newdata)
+  }
+  pp <- brms::prepare_predictions(fit, ...)
+  # Every brms model has the parameter mu
+  lp_mu <- brms::get_dpar(pp, dpar = "mu", inv_link = TRUE)
+  df <- expand.grid(
+    "index" = seq_len(nrow(lp_mu)),
+    "draw" = seq_len(ncol(lp_mu))
+  )
+  df[["mu"]] <- as.vector(lp_mu)
+  for (dpar in setdiff(names(pp$dpars), "mu")) {
+    lp_dpar <- brms::get_dpar(pp, dpar = dpar, inv_link = TRUE)
+    df[[dpar]] <- as.vector(lp_dpar)
+  }
+  class(df) <- c(
+    class(df), paste0(sub(".*_", "", fit$family$name), "_samples")
+  )
+  dt <- as.data.table(df)
+  dt <- add_mean_sd(dt)
```

Comment:
do we want to call this within this function? or have the user do it in post? I am open to an argument for either side

Reply:
I think postprcessing plotting etc will be easier if we leave it here (as we will know it is part of family_samples) but in terms of a function should do one thing I see the arg that users should call this themselves if they want to

Reply:
I'm weakly in favour of having it included here. It could be possible that there are workflows we stop users from doing by not expecting them to use the `add_mean_sd` function much e.g. adding the mean and SD of summaries rather than draws. I also don't think it's much of a downside to have these additional columns

Reply:
Check if returning the same thing and not adding columns.

Reply:
```
#' @export
add_mean_sd.default <- function(data, ...) {
  cli::cli_inform(c(
    "!" = "Natural scale mean and standard deviation parameter columns not
    added: no method available for this family",
    "Consider submitting an issue to https:/github.com/epinowcast/epidist"
  ))
  return(data)
}
```